testing/jormungandr-integration-tests/src/jormungandr/bft/counter.rs

```diff
@@ -0,0 +1,66 @@
+use chain_impl_mockchain::{chaintypes::ConsensusType, testing::WitnessMode};
+use jormungandr_testing_utils::testing::{
+    jormungandr::ConfigurationBuilder, startup, FragmentSender, FragmentSenderSetup,
+};
+
+#[test]
+/// Ensures that blocks with an incorrect signature are rejected by a BFT leader node
+fn parallel_transaction_using_different_lanes() {
+    let receiver = startup::create_new_account_address();
+    let mut sender = startup::create_new_account_address();
+
+    let (jormungandr, _) = startup::start_stake_pool(
+        &[sender.clone()],
+        &[receiver.clone()],
+        ConfigurationBuilder::new()
+            .with_block0_consensus(ConsensusType::Bft)
+            .with_slots_per_epoch(20)
+            .with_block_content_max_size(100000.into())
+            .with_slot_duration(2),
+    )
+    .unwrap();
+
+    let mut fragment_sender = FragmentSender::new(
+        jormungandr.genesis_block_hash(),
+        jormungandr.fees(),
+        jormungandr.default_block_date_generator(),
+        FragmentSenderSetup::no_verify(),
+    );
+
+    let mut checks = vec![];
+
+    fragment_sender = fragment_sender.witness_mode(WitnessMode::Lane(1));
```

Comment:
Unrelated to this pr, but I see now that `WitnessMode::Lane(..)` is a bit confusing, I would prefer something like an explicit `WitnessMode::Account { lane: 3 }`